Dangerfile

```diff
@@ -19,6 +21,87 @@ if (has_new_policy_template.length != 0) && missing_doc_changes
   fail "A README.md is required for new templates"
 end
 
+# checks for broken links in the any file
+# exclude api and auth hostnames
+exclude_hosts = [
+  'api.loganalytics.io',
+  'management.azure.com',
+  'login.microsoftonline.com',
+  'oauth2.googleapis.com',
+  'www.googleapis.com'
+]
+changed_files.each do |file|
+ diff = git.diff_for_file(file)
+ regex =/(^\+).+?(http|https):\/\/[a-zA-Z0-9.\/?=_-]*.+/
```

Comment:
regex can probably just be /(^\+)/ to get newly added/modified lines, let URI.extract do all the work.

Reply:
good feedback.  I made the change.